src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { Todo, FilterType } from '../../types';
+
+interface Props {
+  todos: Todo[];
+  filterType: FilterType;
+  onChangeType: (type: FilterType) => void;
+  onDelete: (todoId: number) => void;
+}
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  filterType,
+  onChangeType,
+  onDelete,
+}) => {
+  const leftTodos = todos.filter(todo => !todo.completed).length;
+  const completedTodos = todos.some(todo => todo.completed);
+
+  function handleDeleteAllCompleted() {
+    const allCompletedTodos = todos.filter(todo => todo.completed);
+
+    allCompletedTodos.map(todo => onDelete(todo.id));
+  }
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {leftTodos} items left
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
```

Comment:
recommend to use map here
for example:
const links = Object.entries(FilterType);
 {links.map(([key, value]) => (
          <a
            href={`#/${value === FilterType.All ? '' : `${value}`}`}
            className={cn('filter__link', { selected: filterType === value })}
            data-cy={`FilterLink${key}`}
            key={key}
            onClick={() => onChangeType(value)}
          >
            {key}
          </a>
        ))}

Reply:
Good idea, i am glad for your opinion.

Thanks!